### PR TITLE
Make next_run return datetime

### PR DESCRIPTION
**Pull request by wangch079**
## Part of https://github.com/elastic/enterprise-search-team/issues/4158

This PR made the following changes:
1. makes `next_run` function return a `datetime` instead of the number of seconds due.
2. separates job scheduling for `on-demand` sync and `scheduled` sync, which means if a job is scheduled to sync soon, and the `sync_now` flag is set as well, one `scheduled` sync and one `on-demand` sync will be created.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/services/job_scheduling.py [hunk 1]
```diff
@@ -94,19 +95,8 @@ async def _schedule(self, connector):
...
-            if connector.sync_now
-            else JobTriggerMethod.SCHEDULED
-        )
-        await self.sync_job_index.create(
-            connector=connector, trigger_method=trigger_method
-        )
-        if connector.sync_now:
-            await connector.reset_sync_now_flag()
+        await self._on_demand_sync(connector)
+        await self._scheduled_sync(connector)
```
Comment (timgrein) [inline connectors/services/job_scheduling.py:99]: Should this also be called `_on_scheduled_sync` for consistency?

**Review by timgrein — APPROVED** [review]
LGTM, only one small nit

File: connectors/services/job_scheduling.py (hunk 1 above)
Comment (artem-shelkovnikov) [reply connectors/services/job_scheduling.py:99]: I think `on_demand` and `scheduled` here make sense - on is not an indicator of "event", but just a part of "on demand" word combination

Comment (wangch079) [reply connectors/services/job_scheduling.py:99]: `on_demand` and `scheduled` are the two values of `JobTriggerMethod` Enum.
https://github.com/elastic/connectors-python/blob/947b8734fc48f3bc760e648487ff1944ba0c6501/connectors/byoc.py#L60-L63

**Pull request merged**